server/test/write_authzmodel.go

```diff
@@ -392,37 +571,107 @@ func TestWriteAuthorizationModel(t *testing.T, datastore storage.OpenFGADatastor
 					},
 				},
 			},
-			err: errors.RelationNotFound("owner", "", nil),
 		},
 		{
-			name: "ExecuteWriteFailsIfTupleToUsersetReferencesUnknownRelation",
+			name: "succeeds on a valid model with a type and type#relation type",
 			request: &openfgapb.WriteAuthorizationModelRequest{
-				TypeDefinitions: &openfgapb.TypeDefinitions{
-					TypeDefinitions: []*openfgapb.TypeDefinition{
-						{
-							Type: "foo",
-							Relations: map[string]*openfgapb.Userset{
-								"writer": {
-									Userset: &openfgapb.Userset_ComputedUserset{
-										ComputedUserset: &openfgapb.ObjectRelation{
-											Object:   "",
-											Relation: "reader",
+				SchemaVersion: "1.1",
+				StoreId:       storeID,
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "group",
+						Relations: map[string]*openfgapb.Userset{
+							"admin":  {Userset: &openfgapb.Userset_This{}},
+							"member": {Userset: &openfgapb.Userset_This{}},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"admin": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type: "user",
 										},
 									},
 								},
+								"member": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type: "user",
```

Comment:
Sorry, did you mean to have `relationref("group", "member")`. 

The test is called "succeeds on a valid model with a type and type#relation type" but I don't see a `type#relation` reference.

I'm assuming you're testing

```
type user

type group
  relations
    define admin: [user]
    define member: [user, group#member]
```
Is that correct?

Also, as a general note, it would be really cool if we could provide the DSL syntax as a string in these tests for validation 😄 That would be so nice!

@rhamzeh this is another opportunity why I feel it would be good for us to leverage WASM with the DSL transformer. If we could pass in a DSL string in our tests and run it through a DSL transformer wasm module loaded in Go which outputs the protobuf model, then our tests like this could be easier to read and understand.

Reply:
It would become:

```
tests := []struct{
  ...
  model string
}{
  {
    ...
    model: `
        type user

        type document
          relations
            define reader: [user]
    `
  },
}
```
instead of writing out the protobuf structs everywhere.

Reply:
If you look a bit more down the `document` type relations have `group#member` in them.

Reply:
Yeah, sorry I missed that. That's why I deleted the first part of my comment 😄 . Sorry about that. The comment about the DSL still applies though. That'd be cool to do at a later time.